src/Resources/assets/shop/js/wishlistModal.js

```diff
@@ -0,0 +1,111 @@
+            wishlistH2Class: 'wishlist-confirmation-modal__header--title',
+            wishlistBodyClass: 'wishlist-confirmation-modal__body',
+            wishlistBodyItemClass: 'wishlist-confirmation-modal__body--input',
+            wishlistConfirmClass: 'wishlist-confirmation-modal__confirm',
+            wishlistCancelBtnClass: 'wishlist-confirmation-modal__confirm--cancel',
+            wishlistConfirmBtnClass: 'wishlist-confirmation-modal__confirm--perform',
+        };
+        this.actions = actions;
+        this.finalConfig = {...this.defaultConfig, ...config};
+    }
+
+    init() {
+        if (this.config && typeof this.config !== 'object') {
+            throw new Error('BitBag - WishlistsListModal - given config is not valid - expected object');
+        }
+        this._renderModal();
+    }
+
+    _renderModal() {
+        this.modal = this._modalTemplate();
+        this.modal.classList.add('bitbag', 'wishlist-modal-initialization');
+        this._modalActions(this.modal);
+        document.querySelector('body').appendChild(this.modal);
+        this.modal.classList.remove('wishlist-modal-initialization');
+        this.modal.classList.add('wishlist-modal-initialized');
+    }
+
+    _modalTemplate() {
+        const modal = document.createElement('div');
+        modal.innerHTML = `    
+            <form name="${this.finalConfig.wishlistFormName}_save" id="${this.finalConfig.wishlistFormName}" method="post" class=${this.finalConfig.wishlistModalClass}>
+                <header class=${this.finalConfig.wishlistheaderClass}>
+                    <h2 class=${this.finalConfig.wishlistH2Class}>
+                        ${this.finalConfig.headerTitle}
+                    </h2>
+                </header>
+                <section data-bb-target="wishlists" class=${this.finalConfig.wishlistBodyClass}>
+                    ${this.finalConfig.wishlistBodyContent}
+                </section>
+                <section class=${this.finalConfig.wishlistConfirmClass}>
+                    <button type="button" data-bb-action="cancel" class=${this.finalConfig.wishlistCancelBtnClass}>
+                        ${this.finalConfig.cancelText}
+                    </button>
+                    <button type="submit" data-bb-action="perform" id="${this.finalConfig.wishlistFormName}_save" class=${this.finalConfig.wishlistConfirmBtnClass}>
+                        ${this.finalConfig.performText}
+                    </button>
+                </section>
+            </form>
+        `;
+
+        return modal;
+    }
+
+    _modalActions(template) {
+        const cancelBtn = template.querySelector('[data-bb-action="cancel"]');
+        const confirmBtn = template.querySelector('[data-bb-action="perform"]');
+        const input = template.querySelector('[data-bb-target="wishlists"] > [data-bb-target="input"]');
+
+        cancelBtn.addEventListener('click', (e) => {
+            e.preventDefault();
```

Comment:
Instead of e.preventDefault, you can add type button to the button that you don't want to have default action.